src/Microsoft.ServiceFabric.Actors/Runtime/ActorStateProviderHelper.cs

```diff
@@ -225,14 +225,17 @@ internal static IActorStateProvider CreateDefaultStateProvider(ActorTypeInformat
             if (actorTypeInfo.StatePersistence.Equals(StatePersistence.Persisted))
             {
 #if DotNetCoreClr
-                // Use KVSActorStatePRovide for Windows.
                 if (System.Runtime.InteropServices.RuntimeInformation.IsOSPlatform(System.Runtime.InteropServices.OSPlatform.Windows))
                 {
                     stateProvider = new KvsActorStateProvider();
                 }
+                else if (Utility.IsMigrationTarget(new List<Type>() { actorTypeInfo.ImplementationType }))

```

Comment:
If MigrationAttribute[MigrationTarget=true] is set and no explicit state provider is passed, we simply ignore the attribute? We should atleast log a warning for unsupported configurations.

Reply:
Similarly RC as Source is unsupported.

Reply:
Updated to have a warning whenever the migration target is not set as expected(RC service with migration source attribute and KVS service with migration target attribute)